### ISSUE TITLE
Please make tokens consistent

### ISSUE THREAD

**Issue by dtantsur** [tokio-rs/mio#41]
Before attempting to write generic token patch, I looked through git history and realised that you've actually dropped this capability in https://github.com/carllerche/mio/commit/49e99b38eef1ffff1e3971d8159d0642ce4a5f0c. While I'm not feeling confident at all about this decision and hope you'll revisit it one day, currently I suggest making it consistent and dropping generic T argument from Timer and EventLoop. It's of little use without support for tokens in the events and it's extremely confusing. It required me too look through the whole source code (and git history!) to finally understand, why EventLoop accepts type argument like Token, but then uses unrelated type Token that is just uint (and IMO should be dropped in favor of just using uint).

On a related side, I saw discussion in the Rust upstream about using uint and int, can't find link now, but it was considered a bad practice using int and uint in most cases. I think it's better to use specific sized types.

TL;DR I suggest dropping T from Timer and EventLoop and making all tokens just uXX where XX is a size, possibly platform-dependent.
